Bottom: 799b40af70865f86b988228453abb29a10ca12d3
Top:    a74a5506d15c21d020fe76fa3c378863eec2d67a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-10 14:01:42 +0000

Try not using no-holes dicer for drawing polygons











---

diff --git a/src/draw.c b/src/draw.c
index 7a506d0..6eb3823 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -2224,6 +2224,7 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon)
     }
   else if (Polygon->Clipped)
     {
+#if 0
       if (!Polygon->NoHolesValid)
         {
           ComputeNoHoles (Polygon);
@@ -2237,6 +2238,15 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon)
             poly.Clipped = poly.Clipped->f;
           } while (poly.Clipped != Polygon->NoHoles);
         }
+#endif
+        {
+          PolygonType poly = *Polygon;
+          poly.Clipped = Polygon->Clipped;
+          do {
+            DrawPolygonLowLevel (&poly, NULL);
+            poly.Clipped = poly.Clipped->f;
+          } while (poly.Clipped != Polygon->Clipped);
+        }
       /* draw other parts of the polygon if fullpoly flag is set */
       /* NB: No "NoHoles" cache for these */
       if (TEST_FLAG (FULLPOLYFLAG, Polygon))
